### PR TITLE
add span_masked_lm task

### LINKED ISSUE #4301: Add span_masked_lm task
## 🚀 Feature Request
Add a T5-style Span Masked LM unsupervised training task as described in [T5 whitepaper](https://arxiv.org/pdf/1910.10683.pdf) (section 3.1.4).

HuggingFace transformers implementation for reference: [link](https://github.com/huggingface/transformers/blob/master/examples/flax/language-modeling/run_t5_mlm_flax.py)

### Motivation

T5 pre-training objective is different from the existing Fairseq tasks. In order support T5-style Span Masked LM unsupervised training a new task has to be added.

### Pitch

I want a Span Masked LM training task to be available in Fairseq. I'm willing to contribute.

### PR DESCRIPTION
**Pull request by clumsy**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [X] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/main/CONTRIBUTING.md)?
- [X] Did you make sure to update the docs?
- [X] Did you write any new necessary tests?

## What does this PR do?
Fixes #4301.

## PR review
Adding T5-style pretraining task, described in https://arxiv.org/pdf/1910.10683.pdf.
Inspired by the [HuggingFace implementation](https://github.com/huggingface/transformers/blob/e1c153cbaa2f4dc6fa10aec8e3afb38c1b437947/examples/flax/language-modeling/run_t5_mlm_flax.py#L286).

## Did you have fun?
A ton!

### CODE REVIEW

**Review by cbalioglu — APPROVED** [review]
I would advise having a review from someone familiar with the details of the T5 model, but technically looks good to me. Approving to unblock.
